engine/proto/master.proto

```diff
@@ -13,7 +13,23 @@ import "engine/proto/resources.proto";
 option go_package = "github.com/pingcap/tiflow/engine/enginepb";
 
 service Discovery {
-    rpc RegisterExecutor(RegisterExecutorRequest) returns(RegisterExecutorResponse) {}
+    rpc RegisterExecutor(RegisterExecutorRequest) returns(Executor) {
+        option (google.api.http) = {
+            post: "/api/v1/executors/register"
+            body: "executor"
```

Comment:
Are we using Open API to register executors?

Reply:
tiflow-operator may use the `ListExecutors` and executor-related API like `OfflineExecutor` later. So I decide to expose `RegisterExecutor` to keep consistent.